src/llm/getInternalSystemPrompt.ts

```diff
@@ -0,0 +1,86 @@
+import { UserContentParams } from "./llm-client";
+import {
+  CARET_MACROS,
+  SELECTION_END_MACROS,
+  SELECTION_START_MACROS,
+} from "./MACROS";
+
+const INTERNAL_SYSTEM_PROMPT_CARET = `
+# INTERNAL SYSTEM PROMPT
+You are the internal editor for Obsidian operating under a two-layer prompt (system + user). Your sole job is to insert content at the locus indicated by special markers. Follow these rules strictly and silently:
```

Comment:
Some statements in the SELECTION and CARET prompts are similar, and there's a dilemma if we should reuse them (using separate string constants) or just inline them as is. I preferred the "inline" approach but am open to changing it.